### ISSUE TITLE
File Parsing/Text Parsing

### ISSUE THREAD

**Issue by alejmedinajr** [alejmedinajr/AI-Detector#38]
This ticket will begin the functionality implementation for file parsing and plaintext parsing. This is mainly because we are already getting the user prompt that is just text, so we need to start looking into how to evaluate the text. We are also beginning the ticket related to file upload, so we can add the functionality for file parsing.

**Comment by alejmedinajr** [comment]
Starting on this now

**Comment by alejmedinajr** [comment]
Current update: I began by first checking maximum prompt lengths, which seems to be around 5000. This means I need to either break the instructions into chunks, or condense it without losing value. I think chunking the instructions would be easier to start with. I also thought of the types of files we would allow, and I looked for a library that can convert pdf file content into text. I am currently implementing a basic method and I will test it with a pdf of my resume. I mainly want to see if the organization of the content in a pdf can result in messed up parsing.

**Comment by alejmedinajr** [comment]
Current update: I spent some time messing with the new file upload feature, (kudos to @Caleb-Highsmith ). I also tested the pretty basic pdf parsing function on one of my cv's, and it seems to have worked as I imagined (see image). I now will look into incorporating this into a prompt, and seeing if the maximum characters are exceeded (and also look into chunking ideas). 
<img width="1512" alt="Screenshot 2024-02-24 at 1 55 48 PM" src="https://github.com/alejmedinajr/AI-Detector/assets/100097809/f4c91685-b2d5-4ab6-a124-3ce0e337c8b7">

**Comment by alejmedinajr** [comment]
Current update: I also added a function that can handle docx file types. With this addition, I also made a function that calls the correct xfiletype_to_text function. This will ensure files are properly converted to text. I did run into an issue with how we currently have things set up. I tried calling the chatgpt function from the file function, but this was not working since it was still "awaiting". I think this means we need to just have functions for obtaining the data from the react app, then have separate functions that handle the different API calls. I think this makes sense because we would have to make the same exact function that waits for user input for every api, but this could be more efficient if we had one function that did this, and several other functions for different APIs.

**Comment by alejmedinajr** [comment]
I am taking a break for now, but will resume soon.

**Comment by alejmedinajr** [comment]
Starting back up on this.

**Comment by alejmedinajr** [comment]
Current update: The functionality for parsing files of the form pdf and docx has been implemented and works without breaking existing features. Some restructuring to the `API.py` file was also done in order to prevent superfluous code and repetition when it comes to getting information from the user. Now, there is one post function for text from the user and one for file uploads. This is likely the only POST functions we need, and we can create regular functions for other functionality. I converted existing AI APIs to work according to this, and tested to make sure functionality was still maintained. I also did not run into any issues with size of text. However, this is still important to look into since some meaning of instruction could be lost for overly large texts (in which case chunking is still needed).
